src/coreclr/src/tools/Common/TypeSystem/Common/MetadataVirtualMethodAlgorithm.cs

```diff
@@ -407,8 +407,6 @@ private static void FindBaseUnificationGroup(MetadataType currentType, Unificati
                 FindBaseUnificationGroup(baseType, unificationGroup);
             }
 
-            Debug.Assert(unificationGroup.IsInGroupOrIsDefiningSlot(originalDefiningMethod));
```

Comment:
Ugh, sadly I don't think removing this assert is correct to fix this problem. I believe the algorithm is being a little too enthusiastic about removing methods from the group. The problem I'm concerned about is if there is a 3rd type which derives. It should be able to override either the initial method, or the .override method via name/sig and redirect both methods. I've been debugging this for a while to come up with a suggestion for you here, and I'll need to continue tomorrow.

Reply:
Alright, well, I think the problem is that the logic for separating from the group at line 422 shouldn't be
```
if (nameSigMatchMemberMethod != null)
```
But instead I think it should be 
```
if ((nameSigMatchMemberMethod != null) && (nameSigMatchMemberMethod != memberMethod))
```
The logic is looking for an override, not an exact match. … I think. This code is as confusing as I remember it being.

Reply:
Thank you David. I looked at your proposed fixed, and debugged a bit, and I do agree with your analysis. I'll post an updated to the PR